### ISSUE TITLE
Check failed to run: execution expired

### ISSUE THREAD

**Issue by matelukas** [sensu-plugins/sensu-plugins-http#37]
Hi guys,

I recently updated ruby to 2.3.0 and found that sensu-plugin loader is reporting run failures when http plugin timeouts. Below is message included with sensu alert email.

```
Check failed to run: execution expired, ["/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-plugin-1.2.0/lib/sensu-plugin/check/cli.rb:23:in `write'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-plugin-1.2.0/lib/sensu-plugin/check/cli.rb:23:in `puts'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-plugin-1.2.0/lib/sensu-plugin/check/cli.rb:23:in `puts'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-plugin-1.2.0/lib/sensu-plugin/check/cli.rb:23:in `output'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-plugin-1.2.0/lib/sensu-plugin/cli.rb:29:in `block (2 levels) in <class:CLI>'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-plugins-http-0.4.0/bin/check-http.rb:209:in `rescue in run'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-plugins-http-0.4.0/bin/check-http.rb:204:in `run'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-plugin-1.2.0/lib/sensu-plugin/cli.rb:56:in `block in <class:CLI>'"]

Admin GUI: http://manager.local:8080
Host: manager
Timestamp: 2016-04-28 22:08:40 +1000
Address:  172.17.11.73
Check Name:  check_http_chef
Command:  /opt/sensu/embedded/bin/check-http.rb -u https://chef.local -t 10 -k
Status:  CRITICAL
Occurrences:  3

```

`root@manager:~# /opt/sensu/embedded/bin/gem --version
2.6.3`

`root@manager:~# /opt/sensu/embedded/bin/ruby --version
ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-linux]`

When I am running plugin directly from command line it returns prroper status.

```
root@manager:~# /opt/sensu/embedded/bin/check-http.rb -u https://chef.local -t 10 -k
CheckHttp CRITICAL: Request timed out
```

Has anyone encountered this issue ?

Thanks.

**Comment by jmesquita** [comment]
I've been seeing this problem as well and the seeing the traceback is not fun. I'll try to investigate it further this week

**Comment by matelukas** [comment]
Thanks @jmesquita

**Comment by carldjohnston** [comment]
I submitted PR #69 to resolve this; the trace in the plugin output isn't nice given it's sent along with the event (and in notifications).

Thanks!

**Comment by majormoses** [comment]
We need to address this but putting in PRs that hardcode something that is already available as an option is not a good idea...For now I would suggest that you use the `--timeout` option with what works for you.

**Comment by majormoses** [comment]
checking out master I am not seeing this can you test the same with master?
```
$ time ./bin/check-http.rb -u https://starwars.com
CheckHttp CRITICAL: Request error: Connection refused - connect(2) for "starwars.com" port 443

real	0m0.182s
user	0m0.112s
sys	0m0.004s
18:35:41 [ qa.bravo.cloudcruiser.com ] [  ] [ ~/projects/personal/sensu-plugins-http (git, master) ]
$ time ./bin/check-http.rb -u http://starwars.com
CheckHttp WARNING: 301

real	0m0.166s
user	0m0.096s
sys	0m0.020s
18:35:50 [ qa.bravo.cloudcruiser.com ] [  ] [ ~/projects/personal/sensu-plugins-http (git, master) ]
$ time ./bin/check-http.rb -u https://app.cloudcruiser.com
CheckHttp OK: 200, 11566 bytes

real	0m0.337s
user	0m0.112s
sys	0m0.008s
```

**Comment by majormoses** [comment]
The version on the original issue is quite old, can you try updating and seeing if you are still having issues?

**Comment by carldjohnston** [comment]
Hi @majormoses - doesn't appear in Sensu 0.29.0 - I was testing against 0.26.5

**Comment by majormoses** [comment]
ok, what I was really looking for is the version of the plugin. Before I close can a few people try updating to the most recent version of the gem and report back if they have issues. I don't want to waste time looking at an already solved issues and would rather focus on current issues.

**Comment by carldjohnston** [comment]
I only have 0.4.0 and 2.0.2 on my systems; unfortunately I can't replicate it with either version on either Sensu 0.26.5 or 0.29.0 - maybe the cause is a library that's updated on my system..?

**Comment by majormoses** [comment]
possibly, or it was fixed in 2.0.2 the original report was with `0.4.0` which is quite old. I am gonna close and if its still an issue after using newer plugin please feel free to re-open.
